lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2508,6 +2532,106 @@ class ConvertAtenPermuteOp : public OpConversionPattern<AtenPermuteOp> {
+    RankedTensorType inputType = input.getType().cast<RankedTensorType>();
+    RankedTensorType resultType =
+        typeConverter->convertType(op->getResult(0).getType())
+            .cast<RankedTensorType>();
+    int64_t rank = resultType.getRank();
+    SmallVector<Value> resultShape;
+    Value zero = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+    Value one = rewriter.create<arith::ConstantIndexOp>(loc, 1);
+
+    SmallVector<Value> offsets, strides;
+    strides.resize(inputType.getRank(), one);
+    for (int64_t i = 0; i < rank; i++)
+      strides[i] = one;
+    offsets.resize(inputType.getRank(), zero);
+    if (auto sliceOp = dyn_cast<AtenSliceTensorOp>(op)) {
+      AtenSliceTensorOp::Adaptor adaptor(operands);
+      Value end, start, stepIndex;
+      resultShape = getTensorSizes(rewriter, loc, input);
+      int64_t dim, step;
+      if (!matchPattern(sliceOp.dim(), m_TorchConstantInt(&dim)))
+        return op->emitError("unimplemented: dim is not constant");
+      if (adaptor.start().getType().isa<Torch::NoneType>()) {
+        start = zero;
+      } else {
+        auto temp = castIndexToInt(rewriter, loc, resultShape[dim]);
+        start = toPositiveDimDynamic(rewriter, loc, adaptor.start(), temp);
+        Value cst0 = rewriter.create<arith::ConstantOp>(
+            loc, rewriter.getZeroAttr(temp.getType()));
+        Value predDimGEZero = rewriter.create<arith::CmpIOp>(
+            loc, arith::CmpIPredicate::sge, start, cst0);
+        start = rewriter.create<SelectOp>(loc, predDimGEZero, start, cst0);
+        start = castIntToIndex(rewriter, loc, start);
+      }
+      if (adaptor.end().getType().isa<Torch::NoneType>()) {
+        end = resultShape[dim];
+      } else {
+        auto temp = castIndexToInt(rewriter, loc, resultShape[dim]);
+        end = toPositiveDimDynamic(rewriter, loc, adaptor.end(), temp);
+        end = castIntToIndex(rewriter, loc, end);
+      }
+      if (!matchPattern(sliceOp.step(), m_TorchConstantInt(&step))) {
+        if (adaptor.step().getType().isa<Torch::NoneType>()) {
+          step = 1;
+        } else {
+          return op->emitError("unimplemented: step is not constant");
+        }
+      }
+      // Slice logic: resultSize = floordiv(end - start + step - 1,  step)
+      stepIndex = rewriter.create<arith::ConstantIndexOp>(loc, step);
+      Value len = rewriter.create<arith::SubIOp>(loc, end, start);
+      Value resultSize = rewriter.create<arith::AddIOp>(loc, len, stepIndex);
+      resultSize = rewriter.create<arith::SubIOp>(loc, resultSize, one);
+      resultSize =
+          rewriter.create<arith::FloorDivSIOp>(loc, resultSize, stepIndex);
+      resultShape[dim] = resultSize;
+
+      offsets[dim] = start;
+      strides[dim] =
+          rewriter.create<arith::MulIOp>(loc, strides[dim], stepIndex);
+      Value result = rewriter.create<tensor::ExtractSliceOp>(
+          loc, input, offsets, resultShape, strides);
+
+      if (rank < inputType.getRank()) {
```

Comment:
Ha, I think we need to add a `torch.squeeze` when decomposing select. This can be fixed in a separate PR though.

Reply:
this applies to AtenSliceTensorOp also, specifically the slice that matches the functionality of select ie
x[0] when x = [ [0,1] , [2,3] ]. I suppose we could have a decomposition case for this also

Reply:
For select I think even the result dim is size 1, the dim is till kept.

Reply:
select will always actually reduce dim torch.select(torch.randn(2,3), 0, 1).size()
torch.Size([3])

Reply:
Right, sry I meant slice in the above comment. @Shukla-Gaurav will add squeeze.dim soon so that we can use that in select decomposing rather than handling it here. Can you add a comment saying this is for select and will be removed once we add squeeze.dim?

Reply:
yes